### DISCUSSION TITLE
Why does Ghidra always mark .got section as read-only?

### DISCUSSION THREAD

**Discussion by AminoACID123** [NationalSecurityAgency/ghidra#5552] [Q&A] [answered]
I am running ghidra 10.3 on an ordinary Linux x86 ELF executable. The `readelf` command tells me that the `.got` section has the `W` flag. However, in my ghidra script `currentProgram.getMemory().getBlock(".got").isWrite()` tells me that the `.got` section is not writable. Why?

**Comment by ghidra1** [comment, marked as answer]
The .got should be getting relocations applied such that the pointers may be treated as constant.  This is most easily achieved for the .got section by making it read-only.  Non-constant pointers will cause decompiler to treat any use of them as variable and result in computed calls/jumps instead of referring to the referenced function/data location.

The only reason the original section as loaded is not read-only is because the dynamic linker must be able to update the pointer values in memory in a lazy fashion.  Our ELF Loader synthesizes the dynamic linking during our relocation processing.
